same-tree/jaejeong1.java

```diff
@@ -0,0 +1,44 @@
+
+//  Definition for a binary tree node.
+class TreeNode {
+  int val;
+  TreeNode left;
+  TreeNode right;
+  TreeNode() {}
+  TreeNode(int val) { this.val = val; }
+  TreeNode(int val, TreeNode left, TreeNode right) {
+    this.val = val;
+    this.left = left;
+    this.right = right;
+  }
+}
+
+class Solution {
+
+  public boolean isSameTree(TreeNode p, TreeNode q) {
+    // 풀이: 재귀로 left와 right를 비교하면서 같은지 확인한다.
+    // TC: O(N)
+    // SC: O(N)
+    return dfs(p, q);
+  }
+
+  private boolean dfs(TreeNode p, TreeNode q) {
+    if (p == null || q == null) {
+      return p == q; // 둘 다 null이면 true, 하나만 null이면 false
+    }
+
+    if (p.val != q.val) { // 값이 다르면 false
+      return false;
+    }
+
+    if (p.left != null && !dfs(p.left, q.left)) { // 둘 중 하나만 null인 케이스는 위에서 걸러져, p.left 하나만 null 여부를 체크하면 npe 방지할 수 있다
+      return false; // 둘다 null이 아닐때만 dfs 값비교를 하고, 결과가 false 라면 false 반환하고 중단
+    }
+
+    if (p.right != null && !dfs(p.right, q.right)) {
+      return false;
+    }
+
+    return true; // 값이 다른 케이스가 없으면 true 반환
```

Comment:
```suggestion
    return dfs(p.left, q.left) && dfs(p.right, q.right);
```
선행조건으로 null 체크를 진행하니 이렇게 하면 더 간단해질 것 같습니다~

Reply:
@bky373 좋은 피드백 감사합니다! 오늘 문제풀이때도 유사한 피드백을 받았는데, 말씀주시는대로 간결하게 작성하는 것이 풀이도 설명도 훨씬 쉽네요 :)